enigma-principal/app/src/boot_network/keys_provider_http.rs

```diff
@@ -168,18 +171,39 @@ impl PrincipalHttpServer {
         };
     }
 
-    /// Endpoint for the get_state_keys method
+    /// This function is used to make sure the km is up and running.
+    /// it can be requested via the jsonRPC server using the following command:
+    /// curl -sb -o /dev/null -X POST -d '{"jsonrpc": "2.0", "id": "1", "method": "getHealthCheck", "params": []}' -H "Content-Type: application/json" 127.0.0.1:3040
+    pub fn health_check(epoch_provider: &EpochProvider) -> Value {
+        // Ethereum
+        let contract_signing_address = match epoch_provider.contract.get_signing_address() {
+            Ok(addr) => addr,
+            Err(_) => return Value::Bool(false),
+        };
+        // Enclave
+        let enclave_signing_address: H160 = match esgx::equote::get_register_signing_address(*epoch_provider.eid) {
+            Ok(addr) => addr.into(),
+            Err(_) => return Value::Bool(false),
+        };
```

Comment:
It is cleaner to use `unwrap_or_else(||return Value:Bool(false)` in both `let` expressions.

Reply:
@moriaab I thought so as well (notice that I've done that in the first commits of this PR) but notice that the return is for the function inside the `unwrap_or_else` and it will be assigned to `enclave_signing_address` leading to an error (the types aren't aligned).